File: java/tcp/ClienteTCP.java
import java.io.*;
import java.net.*;

public class ClienteTCP {
    static final int OFFSET = 16;

    public static void main(String[] args) throws IOException {
        String host = "localhost";
        int porta = 5000 + OFFSET;

        try (Socket socket = new Socket(host, porta);
             PrintWriter saida = new PrintWriter(socket.getOutputStream(), true);
             BufferedReader entrada = new BufferedReader(
                     new InputStreamReader(socket.getInputStream()));
             BufferedReader teclado = new BufferedReader(new InputStreamReader(System.in))) {

            System.out.println("[TCP] Conectado ao servidor. Digite 'sair' para encerrar.");
            String linha;
            while (true) {
                System.out.print("> ");
                linha = teclado.readLine();
                if (linha == null) break;
                saida.println(linha);
                System.out.println(entrada.readLine());
                if (linha.equalsIgnoreCase("sair")) break;
            }
        }
    }
}

File: python/tcp/cliente_tcp.py
import socket

OFFSET = 16

HOST = "localhost"
PORTA = 5000 + OFFSET

with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as cliente:
    cliente.connect((HOST, PORTA))
    print("[TCP] Conectado ao servidor. Digite 'sair' para encerrar.", flush=True)
    arquivo = cliente.makefile("r")

    while True:
        try:
            mensagem = input("> ")
        except EOFError:
            break
        cliente.sendall((mensagem + "\n").encode("utf-8"))
        print(arquivo.readline().strip(), flush=True)
        if mensagem.lower() == "sair":
            break
